### PR TITLE
ci: merge staging to master

### PR DESCRIPTION
**Pull request by maxwell-aiden**
This is an automatic PR generated by the CI/CD pipeline. This will be automatically fast-forward merged if successful.

### CODE REVIEW

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 9707554613 for 1dad4d509409eff623c2d266cb5eb2cdfcc38adc

https://github.com/MatrixAI/js-quic/actions/runs/9707554613

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 9756729507 for 9eec5dd1ca547b39e1260d01d29afb32618d80bd

https://github.com/MatrixAI/js-quic/actions/runs/9756729507

**Comment by maxwell-aiden** [comment]
Pipeline Succeeded on 9756729507 for 9eec5dd1ca547b39e1260d01d29afb32618d80bd

https://github.com/MatrixAI/js-quic/actions/runs/9756729507

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 9768469146 for 6e2bce9e53835911795a0e43bc0787e84fbb1f07

https://github.com/MatrixAI/js-quic/actions/runs/9768469146

**Comment by maxwell-aiden** [comment]
Pipeline Succeeded on 9768469146 for 6e2bce9e53835911795a0e43bc0787e84fbb1f07

https://github.com/MatrixAI/js-quic/actions/runs/9768469146

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 9769190516 for 11b9a4f7937704defd975562ac7cda84641ca9b6

https://github.com/MatrixAI/js-quic/actions/runs/9769190516

**Comment by maxwell-aiden** [comment]
Pipeline Succeeded on 9769190516 for 11b9a4f7937704defd975562ac7cda84641ca9b6

https://github.com/MatrixAI/js-quic/actions/runs/9769190516

File: src/utils.ts [hunk 1]
```diff
@@ -554,6 +555,20 @@ function isStreamReset(e: Error): number | false {
...
+/**
+ * Increases the total number of registered event handlers before a node warning is emitted.
+ * In most cases this is not needed but in the case where you have one event emitter for multiple handlers you'll need
+ * to increase the limit.
+ * @param target - The specific `EventTarget` or `EventEmitter` to increase the warning for.
+ * @param limit - The limit before the warning is emitted, defaults to 100000.
+ */
+function setMaxListeners(
+  target: EventTarget | NodeJS.EventEmitter,
+  limit: number = 100000,
```
Comment (CMCDragonkai) [inline src/utils.ts:567]: This should be in the module configuration defaults.

Do not hardcode the constant here.

Propagate the config variable all the way up to PK system config.

Ensure consistent naming across all evented push-pull modules.

@brynblack @tegefaulkes @amydevs

File: src/utils.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ import type {
   StreamId,
 } from './types';
 import dns from 'dns';
+import events from 'events';
```
Comment (CMCDragonkai) [inline src/utils.ts:13]: Please call this `nodeEvents`.

**Pull request merged**